fe/src/main/java/org/apache/doris/clone/DynamicPartitionScheduler.java

```diff
@@ -121,11 +123,118 @@ public void createOrUpdateRuntimeInfo(String tableName, String key, String value
         defaultRuntimeInfo.put(LAST_UPDATE_TIME, DEFAULT_RUNTIME_VALUE);
         defaultRuntimeInfo.put(LAST_SCHEDULER_TIME, DEFAULT_RUNTIME_VALUE);
         defaultRuntimeInfo.put(DYNAMIC_PARTITION_STATE, State.NORMAL.toString());
-        defaultRuntimeInfo.put(MSG, DEFAULT_RUNTIME_VALUE);
+        defaultRuntimeInfo.put(CREATE_PARTITION_MSG, DEFAULT_RUNTIME_VALUE);
+        defaultRuntimeInfo.put(DROP_PARTITION_MSG, DEFAULT_RUNTIME_VALUE);
         return defaultRuntimeInfo;
     }
 
-    private void dynamicAddPartition() {
+    private ArrayList<AddPartitionClause> getAddPartitionClause(OlapTable olapTable, Column partitionColumn, String partitionFormat) {
+        ArrayList<AddPartitionClause> addPartitionClauses = new ArrayList<>();
+        Calendar calendar = Calendar.getInstance();
+        DynamicPartitionProperty dynamicPartitionProperty = olapTable.getTableProperty().getDynamicPartitionProperty();
+        for (int i = 0; i <= dynamicPartitionProperty.getEnd(); i++) {
+            String prevBorder = DynamicPartitionUtil.getPartitionRange(dynamicPartitionProperty.getTimeUnit(),
+                    i, (Calendar) calendar.clone(), partitionFormat);
+            // continue if partition already exists
+            String nextBorder = DynamicPartitionUtil.getPartitionRange(dynamicPartitionProperty.getTimeUnit(),
+                    i + 1, (Calendar) calendar.clone(), partitionFormat);
+            PartitionValue lowerValue = new PartitionValue(prevBorder);
+            PartitionValue upperValue = new PartitionValue(nextBorder);
+            PartitionInfo partitionInfo = olapTable.getPartitionInfo();
+            RangePartitionInfo info = (RangePartitionInfo) (partitionInfo);
+            boolean isPartitionExists = false;
+            Range<PartitionKey> addPartitionKeyRange;
+            try {
+                PartitionKey lowerBound = PartitionKey.createPartitionKey(Collections.singletonList(lowerValue), Collections.singletonList(partitionColumn));
+                PartitionKey upperBound = PartitionKey.createPartitionKey(Collections.singletonList(upperValue), Collections.singletonList(partitionColumn));
+                addPartitionKeyRange = Range.closedOpen(lowerBound, upperBound);
+            } catch (AnalysisException e) {
+                // keys.size is always equal to column.size, cannot reach this exception
+                LOG.error("Keys size is not equal to column size.");
```

Comment:
Do not use `error` level, error means fatal.
And it better to add details in log, such as e.getMessage().

Same suggest to the `getDropPartitionClause()`

Reply:
done